bindings/node.js/kzg.cxx

```diff
@@ -0,0 +1,356 @@
+    return env.Null();
+  };
+
+  FILE* f = fopen(file_path.c_str(), "r");
+
+  if (f == NULL) {
+    free(kzg_settings);
+    Napi::Error::New(env, "Error opening trusted setup file").ThrowAsJavaScriptException();
+    return env.Null();
+  }
+
+  if (load_trusted_setup(kzg_settings, f) != C_KZG_OK) {
+    free(kzg_settings);
+    Napi::Error::New(env, "Error loading trusted setup file").ThrowAsJavaScriptException();
+    return env.Null();
+  }
+
+  return Napi::External<KZGSettings>::New(info.Env(), kzg_settings);
+}
+
+// freeTrustedSetup: (setupHandle: SetupHandle) => void;
+Napi::Value FreeTrustedSetup(const Napi::CallbackInfo& info) {
+  auto env = info.Env();
+
+  size_t argument_count = info.Length();
+  size_t expected_argument_count = 1;
+  if (argument_count != expected_argument_count) {
+    return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
+  }
+
+  auto kzg_settings = info[0].As<Napi::External<KZGSettings>>().Data();
+  free_trusted_setup(kzg_settings);
+  free(kzg_settings);
+  return env.Undefined();
+}
+
+// blobToKzgCommitment: (blob: Blob, setupHandle: SetupHandle) => KZGCommitment;
+Napi::Value BlobToKzgCommitment(const Napi::CallbackInfo& info) {
+  auto env = info.Env();
+
+  size_t argument_count = info.Length();
+  size_t expected_argument_count = 2;
+  if (argument_count != expected_argument_count) {
+    return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
+  }
+
+  auto blob_param = info[0].As<Napi::TypedArray>();
+  if (blob_param.TypedArrayType() != napi_uint8_array) {
+     return throw_invalid_argument_type(env, "blob", "UInt8Array");
+  }
+  auto blob = blob_param.As<Napi::Uint8Array>().Data();
+
+  auto kzg_settings = info[1].As<Napi::External<KZGSettings>>().Data();
+
+  KZGCommitment commitment;
+  blob_to_kzg_commitment(&commitment, blob, kzg_settings);
+
+  uint8_t commitment_bytes[BYTES_PER_COMMITMENT];
+  bytes_from_g1(commitment_bytes, &commitment);
+  return napi_typed_array_from_bytes(commitment_bytes, BYTES_PER_COMMITMENT, env);
+}
+
+// computeAggregateKzgProof: (blobs: Blob[], setupHandle: SetupHandle) => KZGProof;
+Napi::Value ComputeAggregateKzgProof(const Napi::CallbackInfo& info) {
+  auto env = info.Env();
+
+  size_t argument_count = info.Length();
+  size_t expected_argument_count = 2;
+  if (argument_count != expected_argument_count) {
+    return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
+  }
+
+  auto blobs_param = info[0].As<Napi::Array>();
+  auto kzg_settings = info[1].As<Napi::External<KZGSettings>>().Data();
+
+  auto blobs_count = blobs_param.Length();
+  auto blobs = (Blob*)calloc(blobs_count, sizeof(Blob));
+
+  for (uint32_t blob_index = 0; blob_index < blobs_count; blob_index++) {
+    Napi::Value blob = blobs_param[blob_index];
+    auto blob_bytes = blob.As<Napi::Uint8Array>().Data();
+    memcpy(blobs[blob_index], blob_bytes, BYTES_PER_BLOB);
```

Comment:
this copying is unfortunate - does passing blobs_param more directly not work? (like in the python version). not a big deal if you have to copy though.

Reply:
Agreed. I have tried a bunch of things but can't figure out anything other than memcpy.

`blobs_param` is of type `Napi::Array`. I've been googling and trying things. Nothing so far is working.